### DISCUSSION TITLE
How can I add a border to the inner canvas/editor?

### DISCUSSION THREAD

**Discussion by edwarddubi** [hm21/pro_image_editor#211] [Q&A] [answered]
Quick question. I have my initial transparent canvas set to 500x500, which takes up a bit of the device screen. I would like to add a border or something to help users know the boundary they can edit on. I have checked the editorTheme config and can't find anything to help me do this. Please advise if there is any way this can be done.

Thanks.

**Comment by hm21** [comment]
To help the user identify areas where they can edit the image, I overlay a transparent background that darkens the surrounding layers slightly. This overlay is placed in front of all other layers, and its opacity can be adjusted to control the strength of the effect, as shown below. However, I suggest not setting the opacity value to 1, as this would make it impossible for the user to see any layers positioned outside the image.

```dart
configs: ProImageEditorConfigs(
    imageEditorTheme: ImageEditorTheme(
      /// Default is 0.5
      outsideCaptureAreaLayerOpacity: 0.95,
      /// Change the color of the overlay below
      /// background:
    ),
),
```

For the case you require a complete transparent background as it is in the `movable_background_image` example, you need to wrap the editor inside [CustomPaint](https://github.com/hm21/pro_image_editor/blob/ea4052de35f6f9631216041d90ba754ccd3140d2/example/lib/pages/movable_background_image.dart#L240) and draw there a rect with the bounds of your image.

**Reply by edwarddubi** [reply]
Okay, and sorry I have been busy. I had already tried using the outsideCaptureAreaLayerOpacity prop and didn't give me what I was after.
but I'll try the custom paint to see if I can get what I need and I'll let you know if I succeed. Thanks for quick response.

**Reply by edwarddubi** [reply]
@hm21 I tried implementing the custom paint, but it did not work. followed the example code and didnt work. attached is the image of what I am getting vs what I actually want. thanks
<img width="280" alt="Screenshot 2024-09-07 at 8 58 46 PM" src="https://github.com/user-attachments/assets/346f444f-f7aa-401d-882e-5b7a83828439">

**Reply by hm21** [reply, marked as answer]
Ah, got it you would like to mark an area on the image where the user should add text, rather than marking the image border, correct? If so, you can customize the editor configurations by adding `customWidgets`, as shown in the example below.

```dart
configs: ProImageEditorConfigs(
    designMode: platformDesignMode,
    customWidgets: ImageEditorCustomWidgets(
      mainEditor: CustomWidgetsMainEditor(
        bodyItems: (editor, rebuildStream) {
          return [
            ReactiveCustomWidget(
              stream: rebuildStream,
              builder: (_) => Center(
                child: Container(
                  width: MediaQuery.of(context).size.width * 0.7,
                  height: MediaQuery.of(context).size.height * 0.7,
                  decoration: BoxDecoration(
                    border: Border.all(
                      color: Colors.red,
                      width: 2,
                    ),
                  ),
                ),
              ),
            ),
          ];
        },
      ),
    ),
  ),
```

Keep in mind that the image will always fit to the width and height, similar to `BoxFit.contain`. So, if you set the canvas to 500x500, the image will automatically adjust to fit the editor's size. If you prefer not to use this behavior, you can still follow the instructions above, but you will need to crop the final image manually and set the size from the image to your device size. In that case I recommend using the [image](https://pub.dev/packages/image) package, which simplifies the process of cropping a specific area from the image.

**Reply by edwarddubi** [reply]
@hm21 Sorry again, I have been busy and just came back. Yes, I did something similar and achieved what I wanted using decodedImageSize coupled with IgnorePointer. Thanks again for such an awesome painting library.
